packages/dev/test/StakeManager.test.ts

```diff
@@ -70,29 +79,64 @@ contract('StakeManager', function ([_, relayManager, anyRelayHub, owner, nonOwne
 
     testStakeNotValid()
 
-    it('should not allow not owner to schedule unlock', async function () {
+    it('should not allow anyone to stake before owner is set', async function () {
       await expectRevert(
-        stakeManager.unlockStake(nonOwner, { from: owner }),
+        stakeManager.stakeForAddress(relayManager, initialUnstakeDelay, {
+          value: initialStake,
+          from: owner
+        }),
         'not owner'
       )
     })
 
-    it('relay managers cannot stake for themselves', async function () {
+    it('should allow manager to set its owner', async function () {
+      const { logs } = await stakeManager.setRelayManagerOwner(owner, { from: relayManager })
+      expectEvent.inLogs(logs, 'OwnerSet', {
+        relayManager,
+        owner
+      })
+    })
+
+    it('should not allow manager to change its owner', async function () {
+      await stakeManager.setRelayManagerOwner(owner, { from: relayManager })
       await expectRevert(
-        stakeManager.stakeForAddress(relayManager, initialUnstakeDelay, {
-          value: initialStake,
+        stakeManager.setRelayManagerOwner(owner, {
           from: relayManager
         }),
-        'relayManager cannot stake for itself'
+        'already owned'
+      )
+    })
+
+    it('should not allow not owner to schedule unlock', async function () {
+      await expectRevert(
+        stakeManager.unlockStake(nonOwner, { from: owner }),
```

Comment:
didn't you mean
```js
stakeManager.unlockStake(relayManager , {from: nonOwner })
```
?

Reply:
yes, but it does not matter technically - all these addresses are not in any kind of relationship. Fixed.